src/asl/synth.ts

```diff
@@ -1894,12 +1894,77 @@ export class ASL {
               return { value: !!argOutput.value, containsJsonPath: false };
             }
           });
+        } else if (ref === String) {
+          const [arg] = expr.args;
+          if (!arg) {
+            return {
+              // String()
+              value: "",
+              containsJsonPath: false,
+            };
+          }
+          return this.evalExprToJsonPathOrLiteral(arg.expr, (argOutput) => {
+            // String(var) or String({ key: var })
+            if (ASLGraph.isJsonPath(argOutput) || argOutput.containsJsonPath) {
+              const temp = this.newHeapVariable();
+              const jsonPathStates = this.normalizeOutputToJsonPath(argOutput);
+              const { jsonPath } = ASLGraph.isJsonPath(jsonPathStates)
+                ? jsonPathStates
+                : jsonPathStates.output;
+
+              return {
+                ...ASLGraph.joinSubStates(arg, jsonPathStates, {
+                  // if this was a literal, we know it was an object, not a string
+                  startState: ASLGraph.isLiteralValue(argOutput)
+                    ? "format"
+                    : "checkString",
+                  states: {
+                    // String("") => ""
+                    checkString: {
+                      Type: "Choice",
+                      Choices: [{ ...ASL.isString(jsonPath), Next: "assign" }],
+                      Default: "format",
+                    },
+                    format: {
+                      Type: "Pass",
+                      Parameters: {
+                        "str.$": `States.JsonToString(${jsonPath})`,
+                      },
```

Comment:
Converting it to a JSON string?

Reply:
This worked as I expected for all of the tests... but maybe I'm missing something... I'll add more tests. I think the edge case will be that most objects just print `[object object]` instead of getting stringified.

Reply:
hmm

![image](https://user-images.githubusercontent.com/289213/187692324-ff914f27-af04-4adb-981b-4cd8a81520fb.png)